ts/twitter/src/app/posts/[id]/_components/post-detail.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { Box, Flex, Avatar, Text, Button } from "@chakra-ui/react";
+import { Post } from "@/lib/models/post";
+import { mockPostCard } from "../mocks/post-card";
+import { usePostDetail } from "./use-post-detail";
+
+export const PostDetail = () => {
+  const post: Post = mockPostCard;
+  const { followDisplayText, toggleFollowState } = usePostDetail();
+
+  return (
+    <Box key={post.id} p={8} borderWidth={1} borderRadius="md">
+      <Flex>
+        <Avatar size="md" name={post.user_id} mr={2} />
+        <Text fontWeight="bold" fontSize="xl" mr={4} mt={2}>
+          {post.user_id}
+        </Text>
+        <Button borderRadius="full" onClick={toggleFollowState}>
+          {/* TODO: https://github.com/okuda-seminar/Twitter-Clone/issues/452 
+            - Hide follow button for user's own posts in post detail page. */}
+          {followDisplayText}
+        </Button>
+      </Flex>
+      <Text mt={10} fontSize="xl">
+        {post.text}
+      </Text>
+    </Box>
+  );
+};
```

Comment:
You implemented the logic in an unnecessarily complex way.

```suggestion
export const PostDetail = () => {
  const post: Post = mockPostCard;
  const [isFollowing, setIsFollowing] = useState<boolean>(false);
  const toggleFollowStatus = () => setIsFollowing((prev) => !prev);

  return (
    <Box key={post.id} p={8} borderWidth={1} borderRadius="md">
      <Flex>
        <Avatar size="md" name={post.user_id} mr={2} />
        <Text fontWeight="bold" fontSize="xl" mr={4} mt={2}>
          {post.user_id}
        </Text>
        <Button borderRadius="full" onClick={toggleFollowStatus}>
          {/* TODO: https://github.com/okuda-seminar/Twitter-Clone/issues/452 
            - Hide follow button for user's own posts in post detail page. */}
          {isFollowing ? "Following" : "Follow"}
        </Button>
      </Flex>
      <Text mt={10} fontSize="xl">
        {post.text}
      </Text>
    </Box>
  );
};
```